File: deploy/_probe_users.php
<?php
require __DIR__.'/../vendor/autoload.php';
$app = require __DIR__.'/../bootstrap/app.php';
$app->make(Illuminate\Contracts\Console\Kernel::class)->bootstrap();

echo "workspaces=".App\Models\Workspace::count().PHP_EOL;
foreach (App\Models\Workspace::query()->get() as $ws) {
  echo "WS id={$ws->id} name={$ws->name} users=".$ws->users()->count()." active=".$ws->users()->wherePivot('status','active')->count().PHP_EOL;
  foreach ($ws->users()->get() as $u) {
    echo "  id={$u->id} name={$u->name} email={$u->email} role={$u->pivot->role} status={$u->pivot->status} ext=".($u->pivot->morpheus_extension_num ?? '').PHP_EOL;
  }
}
echo "all_users=".App\Models\User::count().PHP_EOL;
foreach (App\Models\User::orderBy('id')->limit(80)->get() as $u) {
  echo "U id={$u->id} name={$u->name} email={$u->email}".PHP_EOL;
}
if (Illuminate\Support\Facades\Schema::hasTable('workspace_user')) {
  $rows = Illuminate\Support\Facades\DB::table('workspace_user')->orderBy('user_id')->get();
  echo "workspace_user_rows=".$rows->count().PHP_EOL;
  foreach ($rows as $r) {
    echo "  wu ws={$r->workspace_id} user={$r->user_id} role={$r->role} status={$r->status} ext=".($r->morpheus_extension_num ?? '').PHP_EOL;
  }
}
$ws = App\Models\Workspace::query()->orderBy('id')->first();
$roster = app(App\Services\Communications\CommunicationsAgentService::class)->listMonitorableDirectory($ws);
echo "monitorable_roster=".count($roster).PHP_EOL;
foreach ($roster as $a) {
  echo " roster id={$a['user_id']} name={$a['name']} role={$a['role']} ext=".($a['morpheus_extension_num'] ?? '').PHP_EOL;
}

File: deploy/_probe_users.py
#!/usr/bin/env python3
import os, sys
from pathlib import Path
ROOT = Path(__file__).resolve().parents[1]
sys.path.insert(0, str(ROOT))
os.environ.setdefault("DEPLOY_PASSWORD", "balitech1")
import deploy._ssh as m
m.HOST="203.215.161.236"; m.USER="ateg"; m.PASSWORD="balitech1"; m.REMOTE_APP="/var/www/apexone"
from deploy._ssh import connect, sudo_run

PHP = r"""<?php
require '/var/www/apexone/vendor/autoload.php';
$app = require '/var/www/apexone/bootstrap/app.php';
$app->make(Illuminate\Contracts\Console\Kernel::class)->bootstrap();
$ws = App\Models\Workspace::where('name','ApexPayments')->first() ?: App\Models\Workspace::first();
echo "ws={$ws->id} {$ws->name} admin_id={$ws->admin_id}\n";
foreach ($ws->users()->orderBy('users.id')->get() as $u) {
  echo $u->id."\t".$u->name."\t".$u->email."\t".($u->pivot->role??'')."\ttl=".($u->pivot->team_lead_user_id??'')."\tstatus=".($u->pivot->status??'')."\n";
}
"""
ssh=connect()
try:
  sftp=ssh.open_sftp()
  with sftp.file('/tmp/apex_list_users.php','w') as f: f.write(PHP)
  sftp.close()
  print(sudo_run(ssh,'php /tmp/apex_list_users.php',check=False))
finally:
  ssh.close()
